runtime/vm/MHInterpreter.cpp

```diff
@@ -1600,9 +1586,18 @@ VM_MHInterpreter::foldForFoldArguments(j9object_t methodHandle)
 	 			*spCombinerSlot = *(spFirstFoldArgSlot - argumentTypeSlots - 1);
 	 		}
 		}
-		_currentThread->sp = spCombinerSlot;
+		/* after loop spCombinerSlot is at the top of the combiner arguments */
+		Assert_VM_true(spCombinerSlot == _currentThread->sp);
 	}
 
+	/* [... foldHandle args descriptionBytes MethodTypeFrame foldHandle PlaceHolderFrame foldHandle combinerArgs] */
+	insertPlaceHolderFrame(combinerArgSlots, methodHandle, J9VMJAVALANGINVOKEMETHODHANDLE_FOLDHANDLEPLACEHOLDER_METHOD(_vm));
+
+	/* insertPlaceHolderFrame will replace combinerHandle with methodHandle. This should be undone so the stack is correctly 
```

Comment:
This comment isn't correct any more as the MH isn't written until the end.  Can you update it here and in the other case as well?

Reply:
updated.